src/streamlink/plugins/eltrecetv.py

```diff
@@ -13,27 +12,31 @@ def can_handle_url(cls, url):
         return cls._url_re.match(url)
 
     def _get_streams(self):
-
-        video_url_found_hls = ""
-
         if "eltrecetv.com.ar/vivo" in self.url.lower():
-            http.headers = {'Referer': self.url,
-            'User-Agent': "ArtearPlayer/3.2.42 (Linux;Android 4.4.2) ExoPlayerLib/1.5.9"}
-            video_url_found_hls = "http://stream.eltrecetv.com.ar/live13/13tv/13tv1/playlist.m3u8"
+            try:
+                http.headers = {'Referer': self.url,
+                'User-Agent': "ArtearPlayer/3.2.42 (Linux;Android 4.4.2) ExoPlayerLib/1.5.9"}
+                res = http.get('https://api.iamat.com/metadata/atcodes/eltrece')
+                live_search = res.text
```

Comment:
I think using `http.json` would be better for parsing the API response rather than using substring - it will be more reliable in the event of small changes to the response. It will also be easier for other people to understand, which is important for future maintenance :)